### DISCUSSION TITLE
Batch query with different filtering criteria, and control the number of threads

### DISCUSSION THREAD

**Discussion by yushangdi** [milvus-io/milvus#30198] [Q&A and General discussion] [answered]
1. I know that I can do filtered search like below:

points.search(
                   queries, "embeddings",  search_params, limit=TOP_K, expr=expr
                )
                
 But what if I have different filtering criteria for each query in `queries`? Do I have to do point.search() once for each query, or is there a way to batch query it? 
 
 2. How to control the number of threads used in index construction and querying? I saw a previous post about setting `omp_num_threads` in config file, but that's for version 1.1.1. Does it still work for the current 2.3.x version?
 
 I'm doing something like this, is this correct?
 
   standalone:
    container_name: milvus-standalone
    image: milvusdb/milvus:v2.3.4
    command: ["milvus", "run", "standalone"]
    security_opt:
    - seccomp:unconfined
    environment:
      ETCD_ENDPOINTS: etcd:2379
      MINIO_ADDRESS: minio:9000
      **OMP_NUM_THREADS: 16**

**Comment by yhmo** [comment]
1. The definition of the collection.search() is:
```
def search(
        self,
        data: List,
        anns_field: str,
        param: Dict,
        limit: int,
        expr: Optional[str] = None,
        partition_names: Optional[List[str]] = None,
        output_fields: Optional[List[str]] = None,
        timeout: Optional[float] = None,
        round_decimal: int = -1,
        **kwargs,
    ):
```
We can see the "data" is a List type, it is "a list of vectors". You can input multiple embeddings like this:
`collection.search(data = [vector1, vector2, ...., vectorN], anns_field="embeddings", param=search_params, limit=TOP_K, expr=expr)`

"expr" is the filtering criteria. It depends on your business to use it. You can use "and" or "or" to build your filtering criteria.
https://milvus.io/docs/boolean.md#Usage

2. thread number cannot be configured in v2.3.x

**Reply by yushangdi** [reply]
Thank you, but what if I have a different `expr` for each query data in the data list? Is there still a way to do it in batch?

**Reply by yhmo** [reply, marked as answer]
Then you should call search() respectively.
```
collection.search(data = [vector1], anns_field="embeddings", param=search_params, limit=TOP_K, expr=expr1)
collection.search(data = [vector2], anns_field="embeddings", param=search_params, limit=TOP_K, expr=expr2)
collection.search(data = [vector3], anns_field="embeddings", param=search_params, limit=TOP_K, expr=expr3)
```

**Reply by yushangdi** [reply]
I see. Thank you!
